### ISSUE TITLE
Support for V8 Javascript Engine's Math.Random()?

### ISSUE THREAD

**Issue by innsR** [altf4/untwister#31]
Would it be possible to support Javascript's Engine (V8) Math.Random() which is also used by node.js?
(https://github.com/v8/v8-git-mirror/blob/master/src/math.js) (https://code.google.com/p/chromium/codesearch#chromium/src/v8/src/v8.cc&q=MWC&sq=package:chromium&l=146)

Thanks.

**Comment by moloch--** [comment]
On the subject as well:
https://jonasnick.github.io/blog/2015/07/08/exploiting-csgojackpots-weak-rng/

**Comment by altf4** [comment]
Nodejs is too popular to ignore, we should really have support for this. I'll try to take it up after defcon.

**Comment by altf4** [comment]
The line...

```
  return (x < 0 ? (x + 0x100000000) : x) * 2.3283064365386962890625e-10;
```

...bothers me because it's floating point multiplication. But I'll see what I can do to replicate it...

**Comment by innsR** [comment]
Yeah I whipped up a quick and dirty brute-forcer a while ago in C++ and had the same issue and in the end I just kept it simple and dealt with everything in integers and then did the floating point conversion in Javascript with the browser manually which wasn't ideal but I got the end result I was after.

**Comment by altf4** [comment]
Yea, that's what I'm afraid of. For untwister, we can't do offloading to another engine like a browser, so I'll just have to figure out the floating point division thing. No big deal.
